### PR TITLE
fix: python version update

### LINKED ISSUE #570: [Bug] python version need update
### Required prerequisites

- [X] I have searched the [Issue Tracker](https://github.com/camel-ai/camel/issues) and [Discussions](https://github.com/camel-ai/camel/discussions) that this hasn't already been reported. (+1 or comment there if it has.)
- [ ] Consider asking first in a [Discussion](https://github.com/camel-ai/camel/discussions/new).

### Motivation

python version need to be updated to 3.9

### Solution

_No response_

### Alternatives

_No response_

### Additional context

_No response_

### LINKED ISSUE #617: [BUG] Github workflow python version conflict
### Required prerequisites

- [X] I have read the documentation <https://camel-ai.github.io/camel/camel.html>.
- [X] I have searched the [Issue Tracker](https://github.com/camel-ai/camel/issues) and [Discussions](https://github.com/camel-ai/camel/discussions) that this hasn't already been reported. (+1 or comment there if it has.)
- [ ] Consider asking first in a [Discussion](https://github.com/camel-ai/camel/discussions/new).

### What version of camel are you using?

0.1.5

### System information

Github workflow

### Problem description

In set up python environment and install dependecies, python version conflict.
https://github.com/camel-ai/camel/actions/runs/9446906338/job/26017544154?pr=571

### Reproducible example code

github workflow auto run.

### Traceback

```pytb
Current Python version (3.8.18) is not allowed by the project (>=3.9.0,<3.12).
```

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by zhiyu-01**
## Description

Update python version to 3.9

## Motivation and Context

close #570 
close #617 

- [x] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by Wendong-Fan — COMMENTED** [review]
thanks @zhiyu-01 , should we also update version in poetry toml file?

**Comment by Wendong-Fan** [comment]
@onemquan pointed out that the fail reason is due to the poetry lock file was not updated. Every time we updated the poetry toml file we need to run `poetry lock` to update the lock file. Thanks for the contribution from @onemquan and @zhiyu-01 !

File: README.md [hunk 1]
```diff
@@ -104,7 +104,7 @@ exit
 Install `CAMEL` from source with conda and pip:
 ```sh
 # Create a conda virtual environment
-conda create --name camel python=3.10
```
Comment (Wendong-Fan) [inline README.md:107]: there's no need to change README file here

**Review by Wendong-Fan — APPROVED** [review]

**Review by onemquan — APPROVED** [review]
LGTM

**Pull request merged**